### PR TITLE
Update release documentation

### LINKED ISSUE #368: Create RELEASE.md document
Had a conversation with @miry about the current release process and what further info we need to port from the readme to a dedicated document with release instructions.

In general, the process is similar to this:
- PREREQUISITE: Shopify employees performing releases should have their workstation configured to have their commits signed (verified). See [Docs](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
- Prepare for the release locally. 
  - Update the CHANGELOG.md with 2 changes 
      1. Add a new version header at the top of the 'unreleased' section
      2. Update links at bottom of changelog
  -  Update any version references in the README to point to the new version
- Commit this locally (and sign the commit) - Commit message should be something like `Release 2.x.x`
- Create a tag locally and sign it - `git tag -s v2.x.x`
- Run `make test-release` locally - this will run tests and final verifications
- Push the release commit to github (either through a PR, or straight to master branch if you have permissions)
- Wait for tests to turn green after the change is on the master branch (merge your PR as necessary)
- Push your local tag to github
- Goreleaser should create the release on Github via Github Actions
- Verify release notes were generated correctly
- [Update Homebrew versions ](https://github.com/Shopify/homebrew-shopify/blob/master/toxiproxy.rb#L9)
   1. Update app_version string
   2. Update hashes for all versions (find the hashes in the checksums.txt from your release notes)
   3. Do a manual smoketest of installing toxiproxy via brew while in the shopify-homebrew repo locally `brew install ./toxiproxy.rb --debug`

### PR DESCRIPTION
**Pull request by neufeldtech**
Resolves #368

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -616,9 +616,4 @@ For example, `shopify_test_redis_master` or `shopify_development_mysql_1`.
 
 ### Release
 
-1. Update `CHANGELOG.md`
-1. Change versions in `README.md`
-1. Create a release commit and a signed tag.
-1. [Bump version for Homebrew](https://github.com/Shopify/homebrew-shopify/blob/master/toxiproxy.rb#L9)
-
-[blog]: https://shopifyengineering.myshopify.com/blogs/engineering/building-and-testing-resilient-ruby-on-rails-applications
+See [RELEASE.md](./RELEASE.md)
```
Comment (neufeldtech) [inline README.md:619]:
```suggestion
See [RELEASE.md](./RELEASE.md)

[blog]: https://shopifyengineering.myshopify.com/blogs/engineering/building-and-testing-resilient-ruby-on-rails-applications
```

**Review by djlebersilvestre — APPROVED** [review]

**Comment by neufeldtech** [comment]
Note: When testing these instructions, I came across a problem with the proposed workflow of creating a pull request for updating the release notes and preparing the release...

When I did this, because github forced me to do a squash and merge, then there was a mismatch between the commit that landed on master and the tag that I had previously created locally.

Options for this workflow are:
- Prepare release and always push release commit directly to master, and push tag directly afterwards
OR
- Create the tag _after_ the merge commit lands on master - however, then this also means that we can't really run the 'make test-release' workflow locally because then it would cause us the need to update our local tag again.

@miry  when you're back from vacation, let's chat on this some more

**Comment by neufeldtech** [comment]
Note: When testing these instructions, I came across a problem with the proposed workflow of creating a pull request for updating the release notes and preparing the release...

When I did this, because github forced me to do a squash and merge, then there was a mismatch between the commit that landed on master and the tag that I had previously created locally.

Options for this workflow are:
- Prepare release and always push release commit directly to master, and push tag directly afterwards
OR
- Create the tag _after_ the merge commit lands on master - however, then this also means that we can't really run the 'make test-release' workflow locally because then it would cause us the need to update our local tag again.

@miry  when you're back from vacation, let's chat on this some more

File: README.md (hunk 1 above)
Comment (miry) [reply README.md:619]: I think we can remove this section at all.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    - [Create Release Commit and Tag](#create-release-commit-and-tag)
+    - [Run Pre-Release Tests](#run-pre-release-tests)
+  - [Submit Release Pull Request](#submit-release-pull-request)
+  - [Push Release Tag](#push-release-tag)
+  - [Verify Github Release](#verify-github-release)
+  - [Update Homebrew versions](#update-homebrew-versions)
+
+## Before You Begin
+
+Ensure your local workstation is configured to be able to [Sign Commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
```
Comment (miry) [inline RELEASE.md:15]:
```suggestion
Ensure your local workstation is configured to be able to [Sign commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
```

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+### Update the [CHANGELOG.md](CHANGELOG.md)
+
+- Add a new version header at the top of the document, just after `# [Unreleased]`
+- Update links at bottom of changelog
+
+### Create Release Commit and Tag
+
+```bash
+git commit -S -m "Release $RELEASE_VERSION"
```
Comment (miry) [inline RELEASE.md:35]: It is a good place to have ENV RELEASE_VERSION initilized in this step.

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+```bash
+git push origin master
+```
+
+## Push Release Tag
+
+- On your local machine again, push your tag to github
+
+```bash
+git push origin v2.x.y
```
Comment (miry) [inline RELEASE.md:55]:
```suggestion
git push origin "v$RELEASE_VERSION"
```

**Comment by miry** [comment]
@neufeldtech Thank you for the doc ❤️

File: RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+## Before You Begin
+
+Ensure your local workstation is configured to be able to [Sign Commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
+
+## Local Release Preparation
+
+### Checkout latest code
+
+```bash
+export RELEASE_VERSION=2.x.y
```
Comment (neufeldtech) [inline RELEASE.md:22]:
```suggestion
```

File: RELEASE.md (hunk 3 above)
Comment (neufeldtech) [inline RELEASE.md:35]:
```suggestion
export RELEASE_VERSION=2.x.y
git commit -S -m "Release $RELEASE_VERSION"
```

File: RELEASE.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+- Add a new version header at the top of the document, just after `# [Unreleased]`
+- Update links at bottom of changelog
+
+### Create Release Commit and Tag
+
+```bash
+export RELEASE_VERSION=2.x.y
+git commit -a -S -m "Release $RELEASE_VERSION"
+git tag -s "v$RELEASE_VERSION" -m "Release $RELEASE_VERSION"
```
Comment (miry) [inline RELEASE.md:37]: @neufeldtech Here is a bit different message.

The tag message should be similar to one in Release.

Example:

```
v2.x.y

- Add release doc (#1231, @neufeldtech)
- Oops I did it again (#321, @miry)
```

**Review by miry — APPROVED** [review]

**Comment by neufeldtech** [comment]
Hmmm, I might have a flaky test....

```go
=== RUN   TestProxyMetricsReceivedSentBytes
    metrics_test.go:36: expected: [toxiproxy_proxy_received_bytes_total{direction="upstream",listener="localhost:0",proxy="test",upstream="upstream"} 5 toxiproxy_proxy_sent_bytes_total{direction="upstream",listener="localhost:0",proxy="test",upstream="upstream"} 5] got: []
```

This test was added in my prometheus metrics PR https://github.com/Shopify/toxiproxy/pull/366

**Pull request merged**